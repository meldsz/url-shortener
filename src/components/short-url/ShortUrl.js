import React, { useState } from 'react';
import { Button } from 'react-bootstrap';

const ShortUrl = (props) => {
    const { longUrl, shortUrl } = props;
    const [isCopied, setIsCopied] = React.useState(false);

    const copyToClipboard = () => {

    }

    return (
        <div className="col display-short-url">
            <div className="shorturl-title">Shortened URL</div>
            <div>
                <span>
                    <a href={longUrl} target="_blank">{shortUrl}</a>
                </span>
                <Button className="copy-btn"
                    variant="outline-secondary"
                    onClick={() => setIsCopied(true)}>
                    {isCopied ? "Copied!" : "Copy"}
                </Button>
            </div>

        </div>
    )
}

export default ShortUrl;