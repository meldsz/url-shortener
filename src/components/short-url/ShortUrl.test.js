import React from 'react';
import { mount } from 'enzyme';
import ShortUrl from './ShortUrl';

const props = {
    longUrl: 'http://originalurl.com',
    shortUrl: 'http://localhost/jagdasd'
};

describe('ShortUrl Component', () => {
    const shortUrlWrapper = mount(<ShortUrl {...props} />);

    it('renders component correctly', () => {
        expect(shortUrlWrapper).toMatchSnapshot();
      });

    it('renders short url in as a link', () => {
        expect(shortUrlWrapper.find('a').text()).toEqual(props.shortUrl);
    });
})