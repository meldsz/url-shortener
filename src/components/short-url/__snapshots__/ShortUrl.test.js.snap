// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortUrl Component renders component correctly 1`] = `ReactWrapper {}`;
